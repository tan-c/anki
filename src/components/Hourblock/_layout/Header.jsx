import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
import moment from 'moment';
import { Map } from 'immutable';
import toastr from 'toastr';

// import { UiActions } from 'utility-redux/common/ui';
import { currentUserSelector } from 'utility-redux/common/user';
import { threeDayMeasurementSelector } from 'utility-redux/hourblock/dailyMeasurement';

export class Header extends React.Component {
  // constructor(props, context) {
  // super(props, context);
  // }

  state = {
    showEyeTimeoutBliking: false,
    currentTime: moment(),
  }

  componentDidMount() {
    this.updateCurrentTime();

    if (!window.isMobile) {
      this.updateWeather();
    }
  }

  componentWillUnmount() {
    clearInterval(this.updateTimeRunner);
  }

  updateCurrentTime = () => {
    this.updateTimeRunner = setInterval(() => {
      this.setState({ // eslint-disable-line react/no-set-state
        showEyeTimeoutBliking: moment().minutes() % 30 >= 29 && moment().seconds() % 2 === 0,
        currentTime: moment(),
      });
    }, 3000);
  }

  updateWeather = () => {
    this.props.updateWeather();
    setInterval(() => {
      this.props.updateWeather();
    }, 1000 * 60 * 60);
  }

  logout = () => {
    // Clear Access Token and ID Token from local storage
    localStorage.removeItem('access_token');
    localStorage.removeItem('expires_at');
    toastr.info('Localstorage cleared, redirecting...');
    location.reload();
  };

  render() {
    const {
      currentUser,
      weatherInfo,
      threeDayMeasurement,
      currentUserImageSrc,
    } = this.props;

    const currentApp = currentUser.hasIn(['config', 'recentApp']) ? currentUser.getIn(['config', 'recentApp']) : '';

    const { currentTime, showEyeTimeoutBliking } = this.state;

    return (
      <header data-role="header" className={`flex-container-row ${showEyeTimeoutBliking && 'bg-orange'}`}>
        <div id="logo" />
        <span className="flex-1 height-50 border-right" />

        {!window.isMobile && (
          <span className="height-50 border-right padding-horizontal-10">
            {threeDayMeasurement.valueSeq().map(item => (
              <span className="font-24 margin-left-10 line-height-50" key={item.get('_id')}>
                {item.get('morningWeight')}
kg -
              </span>
            ))}
          </span>
        )}

        {!window.isMobile && (
          <span className="height-50 border-right padding-horizontal-10">
            <span className="font-24 margin-left-10 line-height-50">
              {parseInt(weatherInfo.getIn(['current', 'temp']), 10)}
              {' '}
-
              {parseInt(weatherInfo.getIn(['next', 'weather', 'temp']), 10)}
°C
            </span>
            <span className="margin-left-5 height-25">
              {weatherInfo.getIn(['city', 'name'])}
            </span>
          </span>
        )}

        {!window.isMobile && (
          <span className="height-50 padding-horizontal-10 line-height-25 text-right">
            <span className="height-25">
欢迎回来,
              {currentUser.get('userName')}
            </span>
            <br />
            <span className="height-25 font-12">{currentUser.get('email')}</span>
          </span>
        )}

        {!window.isMobile && <img alt="pic" className="profile-image" src={currentUser.has('imageUrl') ? currentUser.get('imageUrl') : currentUserImageSrc} />}

        <div id="currentTime" className="border-left">
          {currentTime.format('HH:mm')}
        </div>

        <div
          id="more-icon"
          className="border-left width-60 font-36 bg-green"
          role="button"
          tabIndex="-1"
          onClick={_ => this.logout()}
        >
          H
        </div>
      </header>
    );
  }
}

Header.defaultProps = {
  currentUser: Map(),
  weatherInfo: Map(),
  threeDayMeasurement: Map(),
};

Header.propTypes = {
  // authLock: PropTypes.object.isRequired,
  currentUserImageSrc: PropTypes.string.isRequired,
  currentUser: PropTypes.object,
  weatherInfo: PropTypes.object,
  threeDayMeasurement: PropTypes.object,

  updateWeather: PropTypes.func.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    // authLock: ownProps.authLock,
    currentUser: currentUserSelector(state),
    currentUserImageSrc: state.ui.getIn(['common', 'currentUserImageSrc']),
    weatherInfo: state.ui.getIn(['common', 'api', 'weatherInfo']),
    threeDayMeasurement: threeDayMeasurementSelector(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    updateWeather: _ => dispatch({
      type: 'SET_WEATHER_INFO',
    }),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
