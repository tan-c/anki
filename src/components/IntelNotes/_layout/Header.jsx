import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
// import { bindActionCreators } from 'redux';
import moment from 'moment';
import { Map } from 'immutable';
import toastr from 'toastr';

// import { UiActions } from 'utility-redux/ui';
import { currentUserSelector } from 'utility-redux/user';

export class Header extends React.Component {
  // constructor(props, context) {
  // super(props, context);
  // }

  state = {
    showEyeTimeoutBliking: false,
    currentTime: moment(),
  }

  componentDidMount() {
    this.updateCurrentTime();

    if (!window.isMobile) {
      this.updateWeather();
    }
  }

  componentWillUnmount() {
    clearInterval(this.updateTimeRunner);
  }

  updateCurrentTime = () => {
    this.updateTimeRunner = setInterval(() => {
      this.setState({ // eslint-disable-line react/no-set-state
        showEyeTimeoutBliking: moment().minutes() % 30 >= 29 && moment().seconds() % 2 === 0,
        currentTime: moment(),
      });
    }, 3000);
  }

  updateWeather = () => {
    this.props.updateWeather();
    setInterval(() => {
      this.props.updateWeather();
    }, 1000 * 60 * 60);
  }

  logout = () => {
    // Clear Access Token and ID Token from local storage
    localStorage.removeItem('access_token');
    localStorage.removeItem('expires_at');
    toastr.info('Localstorage cleared, redirecting...');
    location.reload();
  };

  render() {
    const {
      currentUser,
      currentUserImageSrc,
    } = this.props;

    const currentApp = currentUser.hasIn(['config', 'recentApp']) ? currentUser.getIn(['config', 'recentApp']) : '';

    const { currentTime, showEyeTimeoutBliking } = this.state;

    return (
      <header data-role="header" className={`flex-container-row ${showEyeTimeoutBliking && 'bg-orange'}`}>
        <div id="logo" />
        <span className="flex-1 height-50 border-right" />

        {!window.isMobile && (
          <span className="height-50 padding-horizontal-10 line-height-25 text-right">
            <span className="height-25">
欢迎回来,
              {currentUser.get('userName')}
            </span>
            <br />
            <span className="height-25 font-12">{currentUser.get('email')}</span>
          </span>
        )}

        {!window.isMobile && <img alt="pic" className="profile-image" src={currentUser.has('imageUrl') ? currentUser.get('imageUrl') : currentUserImageSrc} />}

        <div id="currentTime" className="border-left">
          {currentTime.format('HH:mm')}
        </div>

        <div
          id="more-icon"
          className="border-left width-60 font-36 bg-blue"
          role="button"
          tabIndex="-1"
          onClick={_ => this.logout()}
        >
          {currentApp.slice(0, 1).toUpperCase()}
        </div>
      </header>
    );
  }
}

Header.defaultProps = {
  currentUser: Map(),
};

Header.propTypes = {
  // authLock: PropTypes.object.isRequired,
  currentUserImageSrc: PropTypes.string.isRequired,
  currentUser: PropTypes.object,

  updateWeather: PropTypes.func.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    // authLock: ownProps.authLock,
    currentUser: currentUserSelector(state),
    currentUserImageSrc: state.ui.getIn(['common', 'currentUserImageSrc']),
    weatherInfo: state.ui.getIn(['common', 'api', 'weatherInfo']),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    updateWeather: _ => dispatch({
      type: 'SET_WEATHER_INFO',
    }),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);
