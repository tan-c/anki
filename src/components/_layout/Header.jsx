import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { bindActionCreators } from 'redux';
import moment from 'moment';
import { Map } from 'immutable';

import {
  Menu, Container, Button, Icon
} from 'semantic-ui-react';
import { currentUserSelector } from 'utility-redux/user';
import { threeDayMeasurementSelector } from 'utility-redux/dailyMeasurement';
import { UiActions } from 'utility-redux/ui';

export class Header extends React.Component {
  // constructor(props, context) {
  // super(props, context);
  // }

  state = {
    showEyeTimeoutBlinking: false,
    currentTime: moment()
  };

  componentDidMount() {
    this.updateCurrentTime();

    if (!window.isMobile) {
      this.updateWeather();
    }
  }

  componentWillUnmount() {
    clearInterval(this.updateTimeRunner);
  }

  updateCurrentTime = () => {
    this.updateTimeRunner = setInterval(() => {
      this.setState({
        // eslint-disable-line react/no-set-state
        showEyeTimeoutBlinking:
          moment().minutes() % 30 >= 29 && moment().seconds() % 2 === 0,
        currentTime: moment()
      });
    }, 3000);
  };

  updateWeather = () => {
    this.props.updateWeather();
    setInterval(() => {
      this.props.updateWeather();
    }, 1000 * 60 * 60);
  }

  render() {
    const {
      isLeftSidebarOn,
      isRightSidebarOn,
      currentUser,
      weatherInfo,
      threeDayMeasurement,
      currentUserImageSrc,
    } = this.props;

    const { currentTime, showEyeTimeoutBlinking } = this.state;

    const currentApp = currentUser.hasIn(['config', 'recentApp']) ? currentUser.getIn(['config', 'recentApp']) : '';

    return (
      <Menu
        fixed="top"
        id="header"
        inverted
        color="blue"
        // inverted
        data-role="header"
        className={`${showEyeTimeoutBlinking && 'bg-orange'}`}
        style={{
          height: 50
        }}
      >

        {!window.isMobile && (
          <span className="height-50 border-right padding-horizontal-10">
            {threeDayMeasurement.valueSeq().map(item => (
              <span className="font-24 margin-left-10 line-height-50" key={item.get('_id')}>
                {item.get('morningWeight')}
                kg -
              </span>
            ))}
          </span>
        )}

        {!window.isMobile && (
          <span
            className="height-50 border-right padding-horizontal-10"
          >
            <span className="font-24 margin-left-10 line-height-50">
              {parseInt(weatherInfo.getIn(['current', 'temp']), 10)}
              {' '}
              -
              {parseInt(weatherInfo.getIn(['next', 'weather', 'temp']), 10)}
              °C
            </span>
            <span className="margin-left-5 height-25">
              {weatherInfo.getIn(['city', 'name'])}
            </span>
          </span>
        )}

        {!window.isMobile && (
          <span className="height-50 padding-horizontal-10 line-height-25 text-right">
            <span className="height-25">
              欢迎回来,
              {currentUser.get('userName')}
            </span>
            <br />
            <span className="height-25 font-12">{currentUser.get('email')}</span>
          </span>
        )}

        <Container>
          {/* {!window.isMobile && (
          <span className="height-50 padding-horizontal-10 line-height-25 text-right">
            <span className="height-25">
              {'欢迎回来,'}
              {currentUser.get('userName')}
            </span>
            <br />
            <span className="height-25 font-12">
              {currentUser.get('email')}
            </span>
          </span>
        )} */}

          {!window.isMobile && (
            <img
              alt="pic"
              className="profile-image"
              src={
                currentUser.has('imageUrl')
                  ? currentUser.get('imageUrl')
                  : currentUserImageSrc
              }
              style={{
                width: 36,
                borderRadius: 1000,
                height: 36,
              }}
            />
          )}

          <Menu.Menu>
            <Button
              color="blue"
              onClick={(_) => {
                this.props.UiActions.updateIn(['isLeftSidebarOn'], !isLeftSidebarOn);
              }}
            >
              Show Left Menu
            </Button>
            <Button
              color="blue"
              onClick={(_) => {
                this.props.UiActions.updateIn(['isRightSidebarOn'], !isRightSidebarOn);
              }}
            >
              Show Right Menu
            </Button>
          </Menu.Menu>

          <Menu.Item
            position="right"
            name="currentTime"
            style={{
              fontSize: 24,
              height: 50,
              lineHeight: 50,
            }}
          >
            {currentTime.format('HH:mm')}
          </Menu.Item>

          <Menu.Item name="iconButtons" position="right">
            {/* <Button icon>
                <Icon
                  name="graduation cap"
                  color={`${isAnkiModalOn ? 'blue' : 'black'}`}
                  onClick={(_) => {
                    this.props.UiActions.updateIn(['common', 'isAnkiModalOn'], !isAnkiModalOn);
                  }}
                />
              </Button> */}
          </Menu.Item>
        </Container>
      </Menu>
    );
  }
}

Header.defaultProps = {
  isLeftSidebarOn: true,
  isRightSidebarOn: true,

  currentUser: Map(),
  weatherInfo: Map(),
  threeDayMeasurement: Map(),
};

Header.propTypes = {
  isLeftSidebarOn: PropTypes.bool,
  isRightSidebarOn: PropTypes.bool,

  currentUser: PropTypes.object,
  currentUserImageSrc: PropTypes.string.isRequired,
  weatherInfo: PropTypes.object,
  threeDayMeasurement: PropTypes.object,

  updateWeather: PropTypes.func.isRequired,

  UiActions: PropTypes.object.isRequired,
};

function mapStateToProps(state, ownProps) {
  return {
    isLeftSidebarOn: state.ui.getIn(['isLeftSidebarOn']),
    isRightSidebarOn: state.ui.getIn(['isRightSidebarOn']),

    currentUser: currentUserSelector(state),
    currentUserImageSrc: state.ui.getIn(['common', 'currentUserImageSrc']),
    weatherInfo: state.ui.getIn(['common', 'api', 'weatherInfo']),
    threeDayMeasurement: threeDayMeasurementSelector(state),
  };
}

function mapDispatchToProps(dispatch) {
  return {
    removeCurrentUser: currentUser => dispatch({
      type: 'DELETE_USER_SUCCESS',
      user: currentUser,
    }),
    updateWeather: _ => dispatch({
      type: 'SET_WEATHER_INFO',
    }),

    UiActions: bindActionCreators(UiActions, dispatch),
  };
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Header);
