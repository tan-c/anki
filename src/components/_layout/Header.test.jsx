import React from 'react';
import { shallow } from 'enzyme';
import expect from 'expect';
import toJson from 'enzyme-to-json';

// import { Map } from 'immutable';

import { SubHeader } from './Header';

function setup() {
  const props = {
    location: {
      pathname: '',
    },
    selectedProjectId: '',
    edittingTarget: '',

    UserActions: {},
    UiActions: {},
  };

  return shallow(<SubHeader {...props} />);
}

describe('SubHeader Unit Tests', () => {
  describe('when normally set up', () => {
    const wrapper = setup();
    it('renders correctly', () => {
      // const jsonOutput = toJson(wrapper);
      // // expect(jsonOutput).toMatchSnapshot();
      expect(wrapper.find("Menu[data-role='subheader']")).toHaveLength(0);
    });
  });
});
