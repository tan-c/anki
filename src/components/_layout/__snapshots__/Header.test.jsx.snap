// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Unit Tests when normally set up renders correctly 1`] = `
<header
  className="flex-container-row false"
  data-role="header"
>
  <span
    className="flex-1 height-50 border-right"
  />
  <span
    className="height-50 padding-horizontal-10 line-height-25 text-right"
  >
    <span
      className="height-25"
    >
      欢迎回来, 
    </span>
    <br />
    <span
      className="height-25 font-12"
    />
  </span>
  <img
    alt="pic"
    className="profile-image"
    src=""
  />
  <div
    className="border-left"
    id="currentTime"
  >
    00:00
  </div>
  <div
    className="border-left width-60 font-36 bg-blue"
    id="more-icon"
    onClick={[Function]}
    role="button"
    tabIndex="-1"
  >
    AN
  </div>
</header>
`;
